security/integration/utils/kubernetes.go

```diff
@@ -61,15 +65,15 @@ func CreateTestNamespace(clientset kubernetes.Interface, prefix string) (string,
 }
 
 func DeleteTestNamespace(clientset kubernetes.Interface, namespace string) error {
-	if err := clientset.CoreV1().Namespaces().Delete(namespace, &metav1.DeleteOptions{}); err != nil {
+	if err := clientset.CoreV1().Namespaces().Delete(namespace, &metav1.DeleteOptions{GracePeriodSeconds: &immediate}); err != nil {
```

Comment:
Just curious: is the default grace period causing some tests flakey?

Reply:
Yes it does. For the certificate rotation test, secondary pod(istio-ca with short refresh) failed to create a secret sometimes because of previous secret was not removed yet.